test/test.js

```diff
@@ -892,4 +894,40 @@ describe('basic test suite', function () {
     withDom(simpleLight1, test(true))
     withDom(simpleShadow1, test(false))
   })
+
+  describe('unusual selectors', () => {
+    const singleQuote = [{ tagName: 'DIV', classList: ["single'quote"] }]
+    const doubleQuote = [{ tagName: 'DIV', classList: ['double"quote'] }]
+    const unicode = [{ tagName: 'DIV', classList: ['uni👪code'] }]
+    const bracket = [{ tagName: 'DIV', classList: ['[]bracket{}'] }]
+    const hashDot = [{ tagName: 'DIV', classList: ['hash#.dot'] }]
+
+    testSelectors(unusualSelectorsLight1, unusualSelectorsShadow1, [
+      { selector: ".single\\'quote", expected: singleQuote },
+      { selector: "#single\\'quote", expected: singleQuote },
+      { selector: "[data-singlequote=\"single\\'quote\"]", expected: singleQuote },
+      { selector: "[data-singlequote='single\\'quote']", expected: singleQuote },
+
+      { selector: '.double\\"quote', expected: doubleQuote },
+      { selector: '#double\\"quote', expected: doubleQuote },
+      { selector: '[data-doublequote=\'double\\"quote\']', expected: doubleQuote },
+      { selector: '[data-doublequote="double\\"quote"]', expected: doubleQuote },
+
+      { selector: '.uni👪code', expected: unicode },
```

Comment:
I spent some time looking into what the issue could be, particularly with these emojis. AFAICT, it looks like Parsel is only supporting a [small unicode range](https://github.com/LeaVerou/parsel/blob/c4cac5b9574cd47ec54617a3477e22622d8cddd2/parsel.js#L4). This could be improved by adding all possible unicode ranges to the character classes. It's hard for me to wrap my head around all the different [unicode categories, scripts and blocks](http://www.regular-expressions.info/unicode.html#category) and I still don't have a good idea of the level of browser support for eg. Armenian unicode aliases in regexes... I'm sure a unicode expert would know.

My point is, I think we will have to be ok with kagekiri-lite not supporting all possible unicode character ranges, if we want it to be under 10k.

Reply:
Yup, I just wanted us to know what's supported and what's not. For kagekiri-lite, we can make the tests explicitly skip these ones since we know they're not supported.